src/Device/Driver/FANET/Parser.cpp

```diff
@@ -0,0 +1,171 @@
+                              const FanetStation &station) {
+  FanetStation *fanet_slot = stations_list.FindStations(station.address);
+
+  if (fanet_slot == nullptr) {
+    fanet_slot = stations_list.AllocateStation();
+    if (fanet_slot == nullptr)
+      // no more slots available
+      return false;
+
+    fanet_slot->Clear();
+    fanet_slot->address = station.address;
+    fanet_slot->location = station.location;
+    fanet_slot->location_available = station.location_available;
+  }
+
+  fanet_slot->valid.Update(clock);
+  fanet_slot->Update(station);
+
+  return true;
+}
+
+bool
+FanetDevice::IntegrityCheck(char *cmd_str) {
+  for(char *ptr = cmd_str; *ptr != '\0'; ptr++)
+  {
+    if((*ptr >= '0' && *ptr <= '9') ||
+       (*ptr >= 'A' && *ptr <= 'F') ||
+       (*ptr >= 'a' && *ptr <= 'f') ||
+       (*ptr == ',')) {
+      continue;
+    }
+    return false;
+  }
+  return true;
+}
+
+char *
+FanetDevice::RemoveExtraChars(char *cmd_str) {
+  // Remove \r\n and any spaces
+  char *ptr = strchr(cmd_str, '\r');
+  if(ptr == nullptr)
+    ptr = strchr(cmd_str, '\n');
+  if(ptr != nullptr)
+    *ptr = '\0';
+  while(*cmd_str == ' ')
+    cmd_str++;
+  return cmd_str;
+}
+
+/*
+ * Weather:
+ *  #FNF src_manufacturer,src_id,broadcast,signature,type,payloadlength,payload
+ * Return true if packet is processed fully, false otherwise
+ */
+bool
+FanetDevice::ParseRxPacket(char *cmd_str, StationsList &stations_list,
+                           double clock) {
+  // Prepare the line
+  cmd_str = RemoveExtraChars(cmd_str);
+  if ( ! IntegrityCheck(cmd_str))
+    return false;
+
+  FanetStation station;
+  char *start_ptr = (char *)cmd_str;
+
+  // Manufacturer(1 byte)
+  uint8_t manufacturer = strtol(start_ptr, nullptr, 16);
+
+  // Id(2 bytes)
+  start_ptr = strchr(start_ptr, SEPARATOR) + 1;  // comma after src_manufacturer
+  uint16_t id = strtol(start_ptr, nullptr, 16);
+
+  // Assemble address
+  station.address = FanetAddress(manufacturer, id);
+
+  /*
+   * Skip broadcast(1 byte) and signature(2 bytes)
+   *  comma after src_id
+   *  comma after broadcast
+   *  comma after signature
+   */
+  for (int i = 0; i < 3; ++i)
+    start_ptr = strchr(start_ptr, SEPARATOR) + 1;
+
+  // Type(1 byte)
+  station.type = strtol(start_ptr, nullptr, 16);
+
+  // Payload length(1 byte)
+  start_ptr = strchr(start_ptr, SEPARATOR) + 1;  // comma after type
+  station.payload_length = strtol(start_ptr, nullptr, 16);
+
+  /*
+   * Standard limitation of LoRa buffers of 256 bytes and the maximum length
+   * of the FANET MAC header of 11 bytes reduces the permitted payload length
+   * to 245 bytes.
+   */
+  if(station.payload_length > 245)
+    return false;
+
+  // Payload(payload_length bytes)
+  station.payload = new uint8_t[station.payload_length];
+  start_ptr = strchr(start_ptr, SEPARATOR) + 1;  // comma after payloadlength
+  for(int i = 0; i < station.payload_length; i++)
+  {
+    char sstr[3] = {start_ptr[i * 2], start_ptr[i * 2 + 1], '\0'};
```

Comment:
Another null pointer dererefence bug here.

Reply:
Fixed

Reply:
I don't see a fix - what did you change? This still crashes.
(btw. I was wrong - this isn't a nullptr dereference bug, it's a buffer overflow.)